zng/name.go

```diff
@@ -20,6 +25,18 @@ func IsIdentifier(s string) bool {
 	return true
 }
 
+// IsTypeName returns true iff s is a valid zson typedef name (exclusive
+// of integer names for locally-scoped typedefs).
+func IsTypeName(s string) bool {
+	for _, c := range s {
+		if !TypeChar(c) {
+			return false
+		}
+	}
```

Comment:
Are we actually OK with names like "....." or "///" ?

Reply:
The short answer is yes, the syntax allows for odd names just like the record-key syntax does as well (which is the same as JSON's record key syntax).